core/unix/signal.c

```diff
@@ -6037,6 +6020,22 @@ main_signal_handler_C(byte *xsp)
 #endif
 
     default: {
+        /* Handle nudges and suspends, which may use the same signal. */
+        if (sig == SUSPEND_SIGNAL) {
+            if (handle_suspend_signal(dcontext, siginfo, ucxt, frame)) {
+                /* i#1921: see comment above */
+                ASSERT(tr == NULL || tr->under_dynamo_control ||
```

Comment:
Is this assert valid for nudges too?

Reply:
Yes, this is already asserted at line 5739 for all signals but with an extra || sig == SUSPEND_SIGNAL b/c we use that on unknown threads.